posts/using-streamlit-chatgpt-and-snowflake-to-easily-bring-insights-to-end-users.md

```diff
@@ -0,0 +1,660 @@
+
+As we can see from the query returned, ChatGPT has extrapolated column names and the table we'll be running the query against. It also includes an explanation as well as the question we asked it. So here are our first challenges with ChatGPT:
+
+1. How do we provide context to ChatGPT with our user input so it has the columns and other information for the tables we have to query against in Snowflake?
+
+2. How do we ask ChatGPT to only give us the query without any of the additional text?
+
+Challenge Number 2 is pretty easy. When we ask ChatGPT our question, all we need to add at the end is: `Do not include the question in your response. Do not provide explanations.`
+
+Challenge Number 1 is a little bit more difficult. The reason being, is how do we provide context to ChatGPT? Ideally, ChatGPT could use in addition to our database, schema, and table names, the table's columns by name, and their descriptions. If we're building this application to be able to look at all of the databases we use for reporting, it could lead to a large amount of information being provided to ChatGPT for context alone. Since this is a small scale demonstration, I chose to create a YAML file for my tables.
+
+```yaml
+table:
+- name: customer_loyalty
+  schema: raw_customer
+  database: pkrishnan_streamlit_db
+  description: Basic information about the customer
+  columns:
+    - name: customer_id
+      description: Unique customer identifier.
+
+    - name: first_name
+      description: Customer's first name
+
+    - name: last_name
+      description: Customer's last name
+
+    - name: city
+      description: Customer's city of residence
+
+    - name: country
+      description: Customer's country of residence
+
+    - name: postal_code
+      description: Customer's residential postal code
+
+    - name: preferred_language
+      description: Customer's preferred language
+
+    - name: gender
+      description: Customer's gender
+
+    - name: favourite_brand
+      description: Customer's favourite brand
+
+    - name: marital_status
+      description: Customer's marital status
+
+    - name: children_count
+      description: Customer's number of children (if disclosed)
+
+    - name: sign_up_date
+      description: Customer's sign up date
+
+    - name: birthday_date
+      description: Customer's birthday
+
+    - name: e_mail
+      description: Customer's email
+
+    - name: phone_number
+      description: Customer's phone number
+```
+
+I've included only a single table here and put this file, called `table_config.yaml` in a directory called `gpt_sql/config`. Now to provide this context to ChatGPT, all you need to do is read the YAML file in Python and change your message to include the yaml file. We can go ahead and do this in Python after where we setup our Snowflake connection.
```

Comment:
Think it reads more succinctly if you remove both instances of the word "called"

Reply:
Capitalize all instances of YAML